tests/test_run_reference_vmm.py

```diff
@@ -156,3 +180,22 @@ def test_reference_vmm_num_vcpus(kernel):
         expect_vcpus(vmm_process, expected_vcpus)
 
         shutdown(vmm_process)
+
+@pytest.mark.parametrize("kernel", KERNELS_INITRAMFS)
+def test_reference_vmm_mem(kernel):
+    """Start the reference VMM and verify the amount of memory."""
+
+    mem_sizes_mib = [512, 1024, 2048]
```

Comment:
Can you check with a value that is higher that the start of the MMIO Gap address? The value is defined here: https://github.com/rust-vmm/vmm-reference/blob/1c5ed93058ba83b87a18052a8ef2975752932edd/src/vmm/src/lib.rs#L60)

We are creating 2 memory regions when the memory does not fit before the MMIO gap, and we should probably check that we still get the expected amount of memory in that situation.

Reply:
If I'm reading it right, I need a value > 4GB?

That makes sense.  I can also explicitly test the edge cases.  Will fixup.

Reply:
Yap, that's correct. We can remove some other values from the `mem_sizes_mib` list so that the test does not take too long.

Reply:
Ok.  I'm now testing well below, well above, and then right at the start of the MMIO gap to verify that the partitioning logic in `create_guest_memory()` is working:

https://github.com/rust-vmm/vmm-reference/blob/1c5ed93058ba83b87a18052a8ef2975752932edd/src/vmm/src/lib.rs#L222-L230

If that's overkill, I can pull some of it out.

Reply:
I think that's fine for now. In the future we can have some more optimization so the tests take less (like for example grouping together the tests for memory size, vcpu count, and maybe disk?), but for now it's not necessary as we're also still figuring out a testing framework.